internal/cluster/validations/validations.go

```diff
@@ -400,7 +400,19 @@ func ValidateDiskEncryptionParams(diskEncryptionParams *models.DiskEncryption) e
 	if diskEncryptionParams == nil {
 		return nil
 	}
-	if *diskEncryptionParams.Mode == models.DiskEncryptionModeTang {
+	if diskEncryptionParams.EnableOn != nil && (*diskEncryptionParams.EnableOn == models.DiskEncryptionEnableOnAll ||
+		*diskEncryptionParams.EnableOn == models.DiskEncryptionEnableOnMasters ||
+		*diskEncryptionParams.EnableOn == models.DiskEncryptionEnableOnWorkers) &&
+		diskEncryptionParams.Mode == nil {
+		return errors.New("Enabling encryption without specifying mode")
+	}
+	if diskEncryptionParams.EnableOn != nil && *diskEncryptionParams.EnableOn == models.DiskEncryptionEnableOnNone && diskEncryptionParams.Mode != nil {
+		return errors.New("Disabling encryption with specifying mode")
+	}
+	if diskEncryptionParams.Mode != nil && diskEncryptionParams.EnableOn == nil {
+		return errors.New("Setting encryption mode without enabling")
+	}
```

Comment:
I think we can do away with this `if` if you apply the changes I suggested to the previous if.

I'm also ok with being explicit here and keeping this third branch. I guess I would probably move it to the top, then. If anything, for readability purposes (start with the easy checks, walk towards the more complex ones).

Reply:
Done